bootstrap/src/hash.mli

```diff
@@ -27,17 +28,41 @@ module State : sig
       [seed] is set to the equivalent of [hash_fold_string HEMLOCK_ENTROPY
       empty]. *)
 
-  val hash_fold_usize: usize -> t -> t
-  (** [hash_fold_usize u state] incorporates the hash of [u] into [state] and
-      returns the resulting state. *)
+  (** Decomposed hash state generator API for use in implementing [hash_fold]
+      for types that do not map perfectly to other types.  The hash algorithm
+      is the 128-bit variant of {{: https://en.wikipedia.org/wiki/MurmurHash}
+      MurmurHash3}, which hashes 128-bit blocks with automatic support for a
+      partial block during finalization. *)
+  module Gen : sig
+    type outer = t
 
-  val hash_fold_float: float -> t -> t
-  (** [hash_fold_float f state] incorporates the hash of [f] into [state] and
-      returns the resulting state. *)
+    type t
+    (** Hash state generator type. *)
+
+    val init: outer -> t
```

Comment:
It occurred to me while thinking about pseudo-random number generation that `Hash.State.seed` isn't really specific to hashing; we could use that same entropy for PRNG seeding.  What do you think about moving `Hash.state.seed` to, say, `Entropy.seed`, and changing `Hash.State.Gen.init` to take a `u128` rather than `Hash.State.t` parameter?

Reply:
I went ahead and did something close to this.  We now have `Entropy.seed` which is explicitly `u128`, and `Hash.State.seed` remains as an opaque state.  `HEMLOCK_ENTROPY` is now interpreted as a `u128` constant, which has the major benefit that if an application logs the value of `Entropy.seed`, it's possible to reproduce the execution by feeding in the same seed.  Before this change we were string-hashing the contents of `HEMLOCK_ENTROPY`, so even if the application logged the seed, there was no way to feed it back in to a subsequent run.